### PR TITLE
[Magiclysm] Add utility spells

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Mods "[Magiclysm] Add utility spells"

#### Purpose of change

There sure are a lot of combat spells to blow up zombies, but the lore says that literally hundreds of millions of people knew magic before the Cataclysm and 99.99% of them definitely didn't need to cast fireball or magic missile in their daily lives.

#### Describe the solution

Add a few spells

CLASSLESS

- Freshen Up: The single most-commonly-known spell before the Cataclysm and the first choice of all the dabbler mages who only knew a single spell, Freshen Up cleans your teeth, sweetens your breath, takes sweat and stains out of your clothes, exfoliates you, and generally makes you feel your best. [Mechnically, it removes `wet` from all body parts and makes any `FILTHY` clothes you're wearing non-filthy]. Requires water and soap/lye as a component. 

[No spell to clean clothes lying around on the floor because if that spell existed, washing machines would never have been invented]

EARTHSHAPER

- Harden Earth: Turn dirt into stone, either dirt walls or dirt floors.  A good way to increase the resilience of a fortress.
- Stone Slumber: Sink into the earth to sleep. You will heal faster and sleep will be more restorative while you are in the earth. Cast the spell again to return to the surface. This is 100% a test run for something I want to give to XE's Ierde but it fits Earthshapers too. 

KELVINIST

- Extinguish: By transmuting the heat of flames directly into cold, the mage can extinguish an ongoing fire. 
- Flames of Passion: Make yourself more persuasive or more intimidating (but not better at lying--you're too fired up to come up with a good lie). Requires alcohol as a component. 
- Freezer Field: Create a globe of ice that you can place objects into to preserve them.  There's no way to actually freeze the contents (freezing is hardcoded to vehicle parts), but I can simulate it with `spoil_multiplier: 0` on an internal pocket. Sure, it'll be annoying to keep summoning new ice globes and putting things in them--that's why modern people used refrigerators instead of magic to preserve their food.

More to come. 

#### Testing

Extinguish removes fire fields, Harden Earth turns dirt to stone, Freezer Field preserves raw meat (and thus other things) [put raw meet inside and waited 4 days, was still fresh], Flames of Passion increases dialogue success.

Peaceful stone slumber: 
image
Recasting the spell (for 0 mana) gets you out.

### CODE REVIEW

**Comment by Rokharn** [comment]
How about some basic magical classless spells like bright light? Not like one, we have already with candle range, but one like flashlight range of light? Right now only mage armor have bright light effect, if I remember correctly. But using mage armor just as flashlight for daily lives little overkill, imho.

Maybe spell to washing clothes as well. At least it's make sense for regular life to clean clothes magically.

**Comment by Standing-Storm** [comment]
> Maybe spell to washing clothes as well. At least it's make sense for regular life to clean clothes magically.

I actually already had this idea but the implementation depends on how filthy clothes are determined--if the clothes themselves have the FILTHY flag it should be easy

**Comment by RedMisao** [comment]
Adding this just in case it wasn't intentional to exclude them, another two other fire-related fields are `fd_incendiary` and `fd_flame_burst`
PD: I've wanting to make a "fire removal" spell for a long time now but `"effect": "remove_field"` wasn't working (#61633)

**Comment by Standing-Storm** [comment]
> I've wanting to make a "fire removal" spell for a long time now but "effect": "remove_field" wasn't working

Turns out it's as easy as a field `ter_transform` (which the docs do mention is possible but provide no examples of).

**Comment by Standing-Storm** [comment]
> How about some basic magical classless spells like bright light? Not like one, we have already with candle range, but one like flashlight range of light? Right now only mage armor have bright light effect, if I remember correctly. But using mage armor just as flashlight for daily lives little overkill, imho.

To get back to this, I'm not doing this because there's already a technomancer spell that does it (`Lamp`)

**Comment by Rokharn** [comment]
I see. Seems I missed this spell.

How about to add to Freshen Up spell effect with +1 social skill as well of reason 'your smile is dazzling' after cleaning  teeth with magic toothpaste? Most likely person will look much more presentable after this spell at many reasons, looking description.

**Comment by Standing-Storm** [comment]
> How about to add to Freshen Up spell effect with +1 social skill as well of reason 'your smile is dazzling' after cleaning teeth with magic toothpaste? Most likely person will look much more presentable after this spell at many reasons, looking description.

At the moment there's no way to add to social skills (only craft skills) and I wouldn't want to give you a bonus magically that you can't replicate nonmagically--taking an actual shower, brushing your teeth, etc., should give you the same bonus as just doing it magically. Once/if the former is implemented, I'll move the benefits onto Freshen Up as well

> Have idea about druid spell as well. Music of nature. Improves mood, reduces speed (similar to using music player, but not artificial), need to be close to natural areas. The druid listens to the singing of the wind, the chirping of birds, the flow of water and understands with satisfaction that nature is still alive, his work is paying off.

This exists: it's the spell Nature's Commune.

**Comment by Rokharn** [comment]
> This exists: it's the spell Nature's Commune.

It's 'improving your health and calming nearby wildlife. ' Description said nothing about changing caster's mood.

> no way to add to social skills

Relic abilities and aura/equipment. Magiclysm have +throwing skill via relic at similar way.

**Comment by Standing-Storm** [comment]
Huh, interesting! That's not anywhere in the documentation.  I'll have to mess with it and document it.

**Comment by Rokharn** [comment]
Looking at save files, possible values for skills looks-like this (magic_knife_bag as example, social as 'speech' here): 

` "skills": [ { "value": "dodge" }, { "value": "firstaid" }, { "value": "gun" }, { "value": "survival" }, { "value": "chemistry" }, { "value": "archery" }, { "value": "driving" }, { "value": "melee" }, { "value": "pistol" }, { "value": "speech" }, { "value": "swimming" }, { "value": "throw", "add": 3 }, { "value": "bashing" }, { "value": "cutting" }, { "value": "stabbing" }, { "value": "computer" }, { "value": "electronics" }, { "value": "mechanics" }, { "value": "unarmed" }, { "value": "launcher" }, { "value": "rifle" }, { "value": "shotgun" }, { "value": "smg" }, { "value": "cooking" }, { "value": "fabrication" }, { "value": "traps" }, { "value": "spellcraft" }, { "value": "tailor" }, { "value": "metaphysics" } ] `

there is (not sure, but seems so) all other possible implemented ways and most values for relics. 

`"modified_bodyparts": [  ], "mutations": [  ], "values": [ { "value": "ARTIFACT_RESONANCE" }, { "value": "STRENGTH" }, { "value": "DEXTERITY" }, { "value": "PERCEPTION" }, { "value": "INTELLIGENCE" }, { "value": "SPEED" }, { "value": "ATTACK_SPEED" }, { "value": "MOVE_COST" }, { "value": "METABOLISM" }, { "value": "MAX_MANA" }, { "value": "REGEN_MANA" }, { "value": "BIONIC_POWER" }, { "value": "MAX_STAMINA" }, { "value": "REGEN_STAMINA" }, { "value": "MAX_HP" }, { "value": "REGEN_HP" }, { "value": "HUNGER" }, { "value": "THIRST" }, { "value": "FATIGUE" }, { "value": "PAIN" }, { "value": "PAIN_REMOVE" }, { "value": "BONUS_DODGE" }, { "value": "BONUS_BLOCK" }, { "value": "MELEE_DAMAGE" }, { "value": "RANGED_DAMAGE" }, { "value": "ATTACK_NOISE" }, { "value": "SHOUT_NOISE" }, { "value": "FOOTSTEP_NOISE" }, { "value": "SIGHT_RANGE_ELECTRIC" }, { "value": "MOTION_VISION_RANGE" }, { "value": "SIGHT_RANGE_FAE" }, { "value": "SIGHT_RANGE_NETHER" }, { "value": "SIGHT_RANGE_MINDS" }, { "value": "CARRY_WEIGHT" }, { "value": "WEAPON_DISPERSION" }, { "value": "SOCIAL_LIE" }, { "value": "SOCIAL_PERSUADE" }, { "value": "SOCIAL_INTIMIDATE" }, { "value": "SLEEPY" }, { "value": "LUMINATION" }, { "value": "EFFECTIVE_HEALTH_MOD" }, { "value": "MOD_HEALTH" }, { "value": "MOD_HEALTH_CAP" }, { "value": "READING_EXP" }, { "value": "SKILL_RUST_RESIST" }, { "value": "LEARNING_FOCUS" }, { "value": "ARMOR_BASH" }, { "value": "ARMOR_CUT" }, { "value": "ARMOR_STAB" }, { "value": "ARMOR_BULLET" }, { "value": "ARMOR_HEAT" }, { "value": "ARMOR_COLD" }, { "value": "ARMOR_ELEC" }, { "value": "ARMOR_ACID" }, { "value": "ARMOR_BIO" }, { "value": "EXTRA_BASH" }, { "value": "EXTRA_CUT" }, { "value": "EXTRA_STAB" }, { "value": "EXTRA_BULLET" }, { "value": "EXTRA_HEAT" }, { "value": "EXTRA_COLD" }, { "value": "EXTRA_ELEC" }, { "value": "EXTRA_ACID" }, { "value": "EXTRA_BIO" }, { "value": "EXTRA_ELEC_PAIN" }, { "value": "RECOIL_MODIFIER" }, { "value": "ITEM_DAMAGE_PURE" }, { "value": "ITEM_DAMAGE_BASH" }, { "value": "ITEM_DAMAGE_CUT" }, { "value": "ITEM_DAMAGE_STAB" }, { "value": "ITEM_DAMAGE_BULLET" }, { "value": "ITEM_DAMAGE_HEAT" }, { "value": "ITEM_DAMAGE_COLD" }, { "value": "ITEM_DAMAGE_ELEC" }, { "value": "ITEM_DAMAGE_ACID" }, { "value": "ITEM_DAMAGE_BIO" }, { "value": "ITEM_ARMOR_BASH" }, { "value": "ITEM_ARMOR_CUT" }, { "value": "ITEM_ARMOR_STAB" }, { "value": "ITEM_ARMOR_BULLET" }, { "value": "ITEM_ARMOR_HEAT" }, { "value": "ITEM_ARMOR_COLD" }, { "value": "ITEM_ARMOR_ELEC" }, { "value": "ITEM_ARMOR_ACID" }, { "value": "ITEM_ARMOR_BIO" }, { "value": "ITEM_ATTACK_SPEED" }, { "value": "CLIMATE_CONTROL_HEAT" }, { "value": "CLIMATE_CONTROL_CHILL" }, { "value": "COMBAT_CATCHUP" }, { "value": "KNOCKBACK_RESIST" }, { "value": "KNOCKDOWN_RESIST" }, { "value": "FALL_DAMAGE" }, { "value": "FORCEFIELD" }, { "value": "EVASION" }, { "value": "OVERKILL_DAMAGE" }, { "value": "RANGE" }, { "value": "AVOID_FRIENDRY_FIRE" } ], `

File: data/mods/Magiclysm/items/spellbooks.json [hunk 1]
```diff
@@ -606,7 +613,36 @@
...
+    "id": "techno_electrical_work",
+    "type": "BOOK",
+    "category": "manuals",
+    "name": { "str": "The Mage-Engineer's Practical Guide", "str_pl": "copies of The Mage-Engineer's Practical Guide" },
+    "//": "4 Technomancer spells, 2 Stormshaper spells",
+    "description": "This is a heavy, university thaumaturgical-program-level guidebook.  It discusses the various practical applications of magic in repairwork and engineering, with a focus on immediately-usable techniques instead of airy theoretical approaches.",
+    "weight": "4000 g",
+    "volume": "2500 ml",
+    "price": 5000,
+    "material": [ "paper" ],
```
Comment (BrettDong) [inline data/mods/Magiclysm/items/spellbooks.json:631]: This breaks item density test:
```
Mods-(~[slow] ~[.],starting_items)=> ../tests/item_test.cpp:744: FAILED:
Mods-(~[slow] ~[.],starting_items)=>   CHECK( item_density( target ) <= max_density )
Mods-(~[slow] ~[.],starting_items)=> with expansion:
Error: Mods-(~[slow] ~[.],starting_items)=>   1.6f <= 1.2f
Mods-(~[slow] ~[.],starting_items)=> with messages:
Mods-(~[slow] ~[.],starting_items)=>   target.typeId() := string_id( "techno_electrical_work" )
Mods-(~[slow] ~[.],starting_items)=>   target.weight() := 4000000mg
Mods-(~[slow] ~[.],starting_items)=>   target.volume() := 2500ml
```

https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/7702934299/job/20992679360?pr=71339#step:17:210

Comment (Standing-Storm) [reply data/mods/Magiclysm/items/spellbooks.json:631]: Huh. Weird that the test here didn't catch it.

Comment (BrettDong) [reply data/mods/Magiclysm/items/spellbooks.json:631]: The mapgen test failed so item density tests in Magicalysm was never run.

**Comment by Naadn** [comment]
I found you forget to add the `"spell_class": "KELVINIST"` to the Extinguish spell so it became a classless spell

**Comment by LetterShapedGlyphs** [comment]
Harden Earth either needs "IGNORE_WALLS" itself or to cause an aoe effect which ignores walls in order to be able to actually make use of the rammed earth -> solid rock transformation. As it is the effect just ends at the wall and doesn't transform it, but the existing spell is a "line" so I'm not sure how you would want to fix this.

**Pull request merged**